src/core/primitives.cc

```diff
@@ -721,6 +721,24 @@ static std::shared_ptr<AbstractNode> builtin_polygon(const ModuleInstantiation *
   return node;
 }
 
+static std::shared_ptr<AbstractNode> builtin_part(const ModuleInstantiation *inst, Arguments arguments, const Children& children)
+{
+  auto node = std::make_shared<PartNode>(inst);
+
+  Parameters parameters = Parameters::parse(std::move(arguments), inst->location(), {"name"});
+
+  const auto& solid_name = parameters["name"];
+  if (!solid_name.isDefined()) {
+    LOG(message_group::Error, inst->location(), arguments.documentRoot(),
+        "module %1$s() needs a name defined", node->name());
+    return node;
+  }
+  node->solid_name = solid_name.toString();
```

Comment:
We really need a canned argument-check function (perhaps additional arguments to `Parameters::parse()`) that checks for required parameters being present, correct types, et cetera.  [`check_arguments()` in builtin_functions.cc](https://github.com/openscad/openscad/blob/dc2a0d1c28e41016abf4f91ffbc4c6c707c4f64a/src/core/builtin_functions.cc#L79) is a good starting point.

While we're looking at such commonization, it would also be good to have a canned "this is intended to be a leaf node and should not have children" function, replacing the DIY in, e.g., [`builtin_cube()`](https://github.com/openscad/openscad/blob/dc2a0d1c28e41016abf4f91ffbc4c6c707c4f64a/src/core/primitives.cc#L153).